cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -675,6 +615,34 @@ fn assemble_metadata(blob: &[u8], metadata_filename: &str) -> Result<Metadata, E
     }
 }
 
+// Parse the url and returns its components
+fn parse_url(
+    input: &str,
+) -> Result<(String, String, Option<u16>, String), Box<dyn std::error::Error>> {
+    // Check if the URL has a scheme. If not, prepend "dummy://" to it.
```

Comment:
nit: `dummy` isn't [a registered scheme][1], but  [`example`][2] [is][1].

[1]: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
[2]: https://www.rfc-editor.org/rfc/rfc7595.html#section-8